### PR TITLE
irqbalance: systemd service config aligned with upstream

### PR DESCRIPTION
**Pull request by snicket2100**
Aligned systemd service config with the definition in the upstream repo: https://github.com/Irqbalance/irqbalance/blob/master/misc/irqbalance.service#L7.

Other than adding some level of sandboxing it also fixes the "Daemon couldn't be bound to the file-based socket." warning reported on irqbalance startup due to the fact that the "/run/irqbalance" directory didn't exist. The "RuntimeDirectory" property makes sure it gets created. The aforementioned warning didn't cause any problems I could spot though.

I have verified that both `irqbalance` as well as `irqbalance-ui` work fine with this new systemd service config.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
@snicket2100 would you mind, modifying the package to install the package to `$out/lib/systemd/system` and use the `systemd.packages` so that we use the upstream service directly? You can also use `sed` to fix up the path in the ExecStart line.

**Comment by snicket2100** [comment]
> @snicket2100 would you mind, modifying the package to install the package to `$out/lib/systemd/system` and use the `systemd.packages` so that we use the upstream service directly? You can also use `sed` to fix up the path in the ExecStart line.

@Mic92 Sure, sounds great, thanks for the hint, I'm very new to NixOS so I didn't know that approach. Will fix it as soon as I learn how to do it ;)

**Comment by Mic92** [comment]
Just have a look for other users of `systemd.packages` in `./nixos`. The `wantedBy = [ "multi-user.target" ];` option would be still required for instance.

**Comment by snicket2100** [comment]
@Mic92 thanks for the hint once again, does it look good to you now?

File: nixos/modules/services/hardware/irqbalance.nix [hunk 1]
```diff
@@ -13,18 +13,12 @@ in
...
-        path = [ pkgs.irqbalance ];
-        serviceConfig =
-          { ExecStart = "${pkgs.irqbalance}/bin/irqbalance --foreground"; };
-        wantedBy = [ "multi-user.target" ];
-      };
-    };
-
     environment.systemPackages = [ pkgs.irqbalance ];
 
+    systemd.services.irqbalance.wantedBy = ["multi-user.target"];
```
Comment (Mic92) [inline nixos/modules/services/hardware/irqbalance.nix]: This was still needed.

**Pull request merged**